### ISSUE TITLE
Open stack volume can't create with options

### ISSUE THREAD

**Issue by zhitongLBN** [fog/fog#3841]
Open stack volume can't create with options

because in `model/volume` the methods 'save' call `create_volume` like this

```
def save
   requires :name, :description, :size 
   data = service.create_volume(name, description, size, attributes)
```

and in the other side function `create_volume` in `requests/` need to get its options like this:

```
def create_volume(name, description, size, options={}) 
  vanilla_options = ['snapshot_id', 'availability_zone']
  vanilla_options.select{|o| options[o]}.each do |key|
    data['volume'][key] = options[key]
  end
```

so it will never get it's options

**Comment by zhitongLBN** [comment]
I'm up to fix this if you are agree with what I said :)

**Comment by geemus** [comment]
Well, the master version looks like it includes a few more options than that:
```
vanilla_options = [:snapshot_id, :imageRef, :volume_type,
                             :source_volid, :availability_zone, :metadata]
```
But I could see how that could still be a problem. Are there particular other options that are getting dropped?

**Comment by geemus** [comment]
Looks like this is actually fixed by #3844
